sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -54,8 +55,11 @@ private[spark] object HiveUtils extends Logging {
     sc
   }
 
+  private val hive1Version = "1.2.1"
+  private val hive2Version = "2.3.4"
+  val isHive2: Boolean = HiveVersionInfo.getVersion.equals(hive2Version)
```

Comment:
Should this check be a little more general, to match all 2.x versions? should it fail on Hive 3.x?

Reply:
I think it only work on Hive 2.3.x.
1. The current code is not compatible with Hive 2.0 - Hive 2.2: https://github.com/apache/spark/pull/23788/files#diff-53f31aa4bbd9274f40547cd00cf0826dR341
2. The current code is not compatible with Hive 3.1([HIVE-12192](https://issues.apache.org/jira/browse/HIVE-12192)).
```java
[ERROR] /Users/yumwang/spark/sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala:451: type mismatch;
 found   : Timestamp (in org.apache.hadoop.hive.common.type) 
 required: Timestamp (in java.sql) 
[ERROR]             row.setLong(ordinal, DateTimeUtils.fromJavaTimestamp(oi.getPrimitiveJavaObject(value)))
```

Reply:
OK, should this at least check for "isHive23" then and match on starting with "2.3."? otherwise this may well work with 2.3.5 but will fail. What about a future 2.4?

Can we ... drop Hive 1.2.x support entirely here or in a next PR?

Reply:
Yes. `isHive23` is more reasonable. I will update it later.

> Can we ... drop Hive 1.2.x support entirely here or in a next PR?

Removing Hive 1.2.x support may be a bit risky. cc @gatorsmile